pkg/worker/io.go

```diff
@@ -24,7 +24,6 @@ func NewSetupReader(wfm memory.WorkflowMemory, compID string, conditionMap map[i
 }
 
 func (i *setupReader) Read(ctx context.Context) (setups []*structpb.Struct, err error) {
-
 	for idx := range len(i.conditionMap) {
 		setupTemplate, err := i.wfm.GetComponentData(ctx, i.conditionMap[idx], i.compID, memory.ComponentDataSetup)
```

Comment:
Here (and in other places), when I logged the value I got a `BADKEY` string in the output:

```
2024/09/05 14:19:47 INFO +++++ SETUP TEMPLATE !BADKEY="string_value:\"${connection.my-openai}\""
```

However, this doesn't mean the value is not there and the pipeline trigger works as expected. @donch1989 have you faced this when working with the workflow memory?

Reply:
Hi @jvallesm,
I just tried some pipelines, and the setup rendering looks good. Could you share how you're logging this value? I can try again and check from my side.

Reply:
```go
func (i *setupReader) Read(ctx context.Context) (setups []*structpb.Struct, err error) {
	for idx := range len(i.conditionMap) {
		setupTemplate, err := i.wfm.GetComponentData(ctx, i.conditionMap[idx], i.compID, memory.ComponentDataSetup)
		if err != nil {
			return nil, err
		}

		val, _ := setupTemplate.ToStructValue()
		slog.Info("+++++ SETUP TEMPLATE", val) // THIS shows BADKEY

		setupVal, err := recipe.Render(ctx, setupTemplate, i.conditionMap[idx], i.wfm, false)
		if err != nil {
			return nil, err
		}
		setup, err := setupVal.ToStructValue()
		if err != nil {
			return nil, err
		}
		setups = append(setups, setup.GetStructValue())
	}

	return setups, nil
}
```

Reply:
I think the issue is caused by a missing key in the slog. If you use `slog.Info("+++++ SETUP TEMPLATE", "setup", val)`, it should work fine.

Reply:
Oh 🤦 I went quick on the logging and didn't introduce the extra args as key-value. I thought it was the memory store rather than the log itself. Thanks for the eye!